tests/async_engine/test_max_queue_length.py

```diff
@@ -0,0 +1,171 @@
+        "--max-model-len",
+        "1024",
+        "--enforce-eager",
+        "--max-num-seqs",
+        "1",
+        "--max-queue-length",
+        "3",
+        "--max-num-batched-tokens",
+        "2048",
+        "--gpu-memory-utilization",
+        "1"
+    ])
+    ray.get(server_runner.ready.remote())
+    yield server_runner
+    ray.shutdown()
+
+
+@pytest.fixture(scope="session")
+def client():
+    client = openai.AsyncOpenAI(base_url="http://localhost:8000/v1",
+                                api_key="token-abc123",
+                                max_retries=0)
+    yield client
+
+
+@pytest.mark.parametrize("model_name", [MODEL_NAME])
+async def test_max_queue_length(server, client: openai.AsyncOpenAI,
+                                model_name: str):
+    sample_chats = [[{
+        "role": "system",
+        "content": "You are a helpful assistant."
+    }, {
+        "role": "user",
+        "content": "Who won the world series in 2020?"
+    }],
+                    [{
+                        "role": "system",
+                        "content": "You are a helpful assistant."
+                    }, {
+                        "role": "user",
+                        "content": "Where was the 2020 world series played?"
+                    }],
+                    [{
+                        "role": "system",
+                        "content": "You are a helpful assistant."
+                    }, {
+                        "role": "user",
+                        "content": "How long did the 2020 world series last?"
+                    }],
+                    [{
+                        "role": "system",
+                        "content": "You are a helpful assistant."
+                    }, {
+                        "role":
+                        "user",
+                        "content":
+                        "What were some television viewership statistics?"
+                    }],
+                    [{
+                        "role": "system",
+                        "content": "You are a helpful assistant."
+                    }, {
+                        "role": "user",
+                        "content": "Why was the 2020 world series so popular?"
+                    }]]
+
+    async def make_api_call(sample_chat):
+        chat_completion = await client.chat.completions.create(
+            messages=sample_chat,
+            model=model_name,
+            temperature=0.8,
+            presence_penalty=0.2,
+            max_tokens=400,
+        )
+        return chat_completion
+
+    async def main():
+        coroutines = [
+            make_api_call(sample_chat) for sample_chat in sample_chats
+        ]
+
+        responses = await asyncio.gather(*coroutines, return_exceptions=True)
+
+        for response in responses:
+            logger.info(response)
+            if isinstance(response, JSONResponse):
+                assert response.status_code == 503
+
+    await main()
```

Comment:
You are already in sync context. You can make it a lot easier by just doing
```python
coroutines = [ client.chat.completions.create(
            messages=sample_chat,
            model=model_name,
            temperature=0.8,
            presence_penalty=0.2,
            max_tokens=400,
        ) for sample_chat in sample_chats]
responses = asyncio.gather(*coroutines, ...)
for ...
```

Reply:
Also because you are using the server, you should just move the test into `test_openai_server.py` in entrypoints to reuse the same server.